### ISSUE TITLE
Amazon|Fedora Linux: shim doesn't always start on reboot

### ISSUE THREAD

**Issue by jamiesonbecker** [userify/shim#21]
The shim.sh on Amazon Linux doesn't always start on reboot.

Additional details:
* cannot find modifications of rc.local or any start script.
* no shim.py, shim.sh, and/or sleep shows in ps ax
* /var/log/userify_shim.log shows log messages from prior to reboot only

**Comment by jamiesonbecker** [comment]
Cleanest resolution is probably a systemd script, which would also be cleaner for newer versions of Ubuntu, Fedora, Debian, and RHEL.

Sadly, we apparently don't learn from past UNIX mistakes, as there's literally no consistent way (not even a simple flag or file in /etc) to detect if systemd is in use.

https://www.google.com/search?q=detecting+systemd

**Comment by jamiesonbecker** [comment]
Ok, it is modifying /etc/rc.d/rc.local and not /etc/rc.local, per Fedora, but that script never gets executed.. an annoying distro bug from upstream. (Note: this does not affect RHEL/Cent, which work differently). Possible solution: fix the distro bug.

**Comment by jamiesonbecker** [comment]
Also on RHEL6 and Cent6. and on all (rhel6/cent6/amazon) no scripts in /etc/rc.local or /etc/rc.d/rc.local are executed, which is a distribution bug that was fixed in RHEL7 but not in Amazon Linux. These distributions are in widespread use (obviously) so the fix might be to check /etc/redhat-release or similar and install an init.d script. (We've tried to avoid that, since every distribution and even most versions of distributions prefers them to have unique macros/metadata/etc, and it'll quickly turn into an unwieldy mess.)
